### PR TITLE
Trainer

### PR DESCRIPTION
**Pull request by mousyball**
* [fix] Config inheritance in multiple base.
* [feat] Add load_pretrained_weight feature and test

[bug]
'shutil.rmtree' in test_utils still remain 1 directory.
i.e. it cant clean up the working directory entirely.

### CODE REVIEW

File: pytorch_trainer/trainer/utils.py [hunk 1]
```diff
@@ -155,3 +165,33 @@ def create_work_dir(work_dir):
...
+    """
+    weight_path = osp.join(dir_path, weight_name)
+    if not os.path.isfile(weight_path):
+        assert False, "Path to pretrained weight does not exist."
+
+    state_dict = torch.load(weight_path,
+                            map_location=lambda storage, loc: storage)
+
+    # [NOTE] Strict mode means the weight should totally match to the model.
+    unexpected, missing = model.load_state_dict(state_dict, strict=True)
```
Comment (JanLin0817) [inline pytorch_trainer/trainer/utils.py]: Need exception handler here.
If set strict as True, code below line 187 will never be triggered
When state_dict mismatch error will pop out

File: tests/trainer/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        # load pretrained weight
+        if config.trainer.pretrain.load_weight:
+            dir_path = work_dir
+            weight_name = output_model_name
+            load_pretained_weight(model,
+                                  dir_path,
+                                  weight_name,
+                                  logger)
+
+    shutil.rmtree(WORK_DIR)
```
Comment (JanLin0817) [inline tests/trainer/test_utils.py]: Class `TestLoadPretraintedWeight`  create work_dir twice but only delet the last one

**Review by JanLin0817 — COMMENTED** [review]
Please fix the issue

File: tests/trainer/test_utils.py (hunk 2 above)
Comment (mousyball) [reply tests/trainer/test_utils.py]: WORK_DIR is the root folder, so the two additional children folder should be deleted.
But it didnt. Don't know the way what pytest is doing during test.

File: pytorch_trainer/trainer/utils.py (hunk 1 above)
Comment (mousyball) [reply pytorch_trainer/trainer/utils.py]: Resolve by the lastest commit.
* [feat] load weight without strict mode and handle error with assertion.

File: tests/trainer/test_utils.py (hunk 2 above)
Comment (mousyball) [reply tests/trainer/test_utils.py]: Resolve by the latest commit.
* [test] Add rmtree to clean up the generated stuffs.
* [feat] Use pytest.fixture to clean up the tempfile.
   => It cleans up the tempfile after assertion.

**Review by JanLin0817 — COMMENTED** [review]
Good job

**Pull request merged**